### PR TITLE
Use and recognize WordPress.com opt out of analytics setting

### LINKED ISSUE #574: Settings: Sync Tracks Opt Out setting
WPiOS issue Ref. https://github.com/wordpress-mobile/WordPress-iOS/issues/8933, implementation Ref. https://github.com/wordpress-mobile/WordPress-iOS/pull/9275.
 
## Sync global opt-out preference
As part of the GDPR efforts, WordPress.com recently introduced a new setting to opt-out. We want to sync the opt-out in the apps with that value whenever the user is signed in.

The setting is stored as tracks_opt_out in me/settings.

The toggle would still be in app settings. It would show the value for the primary account if there’s one, otherwise the local per-app setting. When the user changes the setting, it would change the local value and for all the WordPress.com accounts in the app.

### PR DESCRIPTION
**Pull request by astralbodies**
Closes #574 

Adds Storage, Networking, and basic Yosemite components to support the Tracks opt-out setting to be synced to the app. 

## Things to note

* The `/me/settings` endpoint does contain a few things that `/me` has but the field names aren't identical. I opted for a new data model for the endpoint in case we need to add additional items down the road.
* I've created a new data model to accommodate the new `AccountSettings` entity and decided to not use a relationship to `Account` for now. 
* User ID isn't returned by the `/me/settings` endpoint so I'm stuffing it into the mapper as a parameter like we've done elsewhere.

## To Test

1. Make sure unit tests run.
2. Make sure upgrading from a previous version succeeds.

### CODE REVIEW

**Comment by astralbodies** [comment]
Ready for review @ctarda & @mindgraffiti.

File: Networking/Networking/Remote/AccountRemote.swift [hunk 1]
```diff
@@ -17,6 +17,22 @@ public class AccountRemote: Remote {
     }
 
 
+    /// Loads the AccountSettings associated with the Credential's authToken.
```
Comment (ctarda) [inline Networking/Networking/Remote/AccountRemote.swift]: I am not sure if this is relevant, but other methods that receive a `userID` as parameter as annotated as `Loads ... with the WordPress.com user`

File: Networking/NetworkingTests/Responses/me-settings.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	"user_login": "test123123",
+	"password": "",
+	"display_name": "test123123",
+	"first_name": "Dem 123",
+	"last_name": "Nines",
+	"description": "🙃\n\nThis is a bit more about me.\n\nAnd multi line text.",
+	"user_email": "moocow@example.com",
+	"user_email_change_pending": false,
+	"new_user_email": "",
+	"user_URL": "http://www.xcode.wtf",
```
Comment (mindgraffiti) [inline Networking/NetworkingTests/Responses/me-settings.json]: 🤨 how about `.mmm` instead of `.wtf` ?

Comment (astralbodies) [reply Networking/NetworkingTests/Responses/me-settings.json]: LOL I had forgotten I had that domain associated with my test account. To be fair it is a real domain and I own it. 😂🤣 I'll change it to something less offensive for sure!!

File: Networking/Networking/Model/AccountSettings.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    }
+}
+
+
+// MARK: - Equatable Conformance
+//
+extension AccountSettings: Equatable {
+
+    public static func == (lhs: AccountSettings, rhs: AccountSettings) -> Bool {
+        return lhs.tracksOptOut == rhs.tracksOptOut
```
[the remark this answers is not in the document]
Comment (astralbodies) [reply Networking/Networking/Model/AccountSettings.swift]: Good catch @ctarda - I added that property later in the game and forgot to come back to add it to this equatable function. I can now also implement Comparable if I so chose to but it's probably not needed at the moment.

File: Networking/Networking/Remote/AccountRemote.swift (hunk 1 above)
Comment (astralbodies) [reply Networking/Networking/Remote/AccountRemote.swift]: Technically the user ID is ignored in this case but we're passing it along for the ride to make it into the Storage layer. We're relying upon the bearer token to provide the magic on the API side to return the settings for the user.

**Comment by astralbodies** [comment]
> Will the POST request be part of another PR?

@mindgraffiti yup I didn't want to get over my head with this PR 😋

**Comment by astralbodies** [comment]
Thanks for the review @ctarda & @mindgraffiti!

**Pull request merged**